src/json-helper.c

```diff
@@ -149,10 +149,22 @@ JsonArray* json_get_array(JsonNode *json_node, const gchar *path,
         return g_steal_pointer(&res_arr);
 }
 
-gboolean json_contains(JsonNode *root, gchar *key)
+gboolean json_contains(JsonNode *json_node, gchar *path)
 {
-        JsonNode *node = json_path_query(key, root, NULL);
-        gboolean result = (node != NULL && json_array_get_length(json_node_get_array(node)) > 0);
-        json_node_unref(node);
-        return result;
+        g_autoptr(GError) error = NULL;
+        g_autoptr(JsonNode) node = NULL;
+
+        g_return_val_if_fail(json_node, FALSE);
+        g_return_val_if_fail(path, FALSE);
+
+        node = json_path_query(path, json_node, &error);
+        if (!node) {
+                g_warning("%s", error->message);
```

Comment:
Maybe add a line

     /* failed to compile expression to JSONPath */

to indicate that the error here is only triggered when the 'path' argument is wrong, but not when the query did not fin anyhting (I myself had to look up that json_path_query() acutally always returns an array).

Reply:
Done.